### PR TITLE
[MI-2451]:Added messaged for CRUD operation for subscription template

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**

### CODE REVIEW

File: server/subscribe.go [hunk 1]
```diff
@@ -1204,6 +1215,15 @@ func (p *Plugin) httpEditSubscriptionTemplates(w http.ResponseWriter, r *http.Re
...
 	}
 
+	_, appErr := p.API.CreatePost(&model.Post{
+		UserId:    p.getConfig().botUserID,
+		ChannelId: subscriptionTemplate.ChannelID,
+		Message:   fmt.Sprintf("Jira subscription template, \"%v\", was updated by %v", subscriptionTemplate.Name, connection.DisplayName),
+	})
+	if appErr != nil {
+		return respondErr(w, http.StatusInternalServerError, errors.WithMessage(appErr, "failed to create notification post"))
+	}
```
Comment (ayusht2810) [inline server/subscribe.go:1225]:
```suggestion
	if _, appErr := p.API.CreatePost(&model.Post{
		UserId:    p.getConfig().botUserID,
		ChannelId: subscriptionTemplate.ChannelID,
		Message:   fmt.Sprintf("Jira subscription template, \"%v\", was updated by %v", subscriptionTemplate.Name, connection.DisplayName),
	}); appErr != nil {
		return respondErr(w, http.StatusInternalServerError, errors.WithMessage(appErr, "failed to create notification post"))
	}
```

File: server/subscribe.go [hunk 2]
```diff
@@ -1229,6 +1249,15 @@ func (p *Plugin) httpCreateSubscriptionTemplate(w http.ResponseWriter, r *http.R
 		return code, err
 	}
 
+	_, appErr := p.API.CreatePost(&model.Post{
+		UserId:    p.getConfig().botUserID,
+		ChannelId: subscriptionTemplate.ChannelID,
+		Message:   fmt.Sprintf("Jira subscription template, \"%v\", was added by %v", subscriptionTemplate.Name, connection.DisplayName),
+	})
+	if appErr != nil {
```
Comment (ayusht2810) [inline server/subscribe.go:1257]: same

File: server/subscribe.go [hunk 3]
```diff
@@ -1261,6 +1300,16 @@ func (p *Plugin) httpDeleteSubscriptionTemplate(w http.ResponseWriter, r *http.R
 		return code, err
 	}
 
+	_, appErr := p.API.CreatePost(&model.Post{
+		UserId:    p.getConfig().botUserID,
+		ChannelId: subscriptionTemplate.ChannelID,
+		Message:   fmt.Sprintf("Jira subscription template, \"%v\", was removed by %v", subscriptionTemplate.Name, connection.DisplayName),
+	})
+	if appErr != nil {
```
Comment (ayusht2810) [inline server/subscribe.go:1308]: same

File: server/subscribe.go [hunk 4]
```diff
@@ -1261,6 +1300,16 @@ func (p *Plugin) httpDeleteSubscriptionTemplate(w http.ResponseWriter, r *http.R
...
+		return respondErr(w, http.StatusInternalServerError,
+			errors.WithMessage(appErr, "failed to create notification post"))
```
Comment (ayusht2810) [inline server/subscribe.go:1310]: we can create a constant for this as it is used multiple times

**Review by ayusht2810 — APPROVED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/subscribe.go [hunk 5]
```diff
@@ -280,6 +281,16 @@ func (p *Plugin) getSubscriptionTemplatesForInstance(instanceID types.ID) (*Temp
 	return subs, nil
 }
 
+func (p *Plugin) getSubscriptionTemplatesByID(instanceID types.ID, templateID string) (*SubscriptionTemplate, error) {
```
Comment (manojmalik20) [inline server/subscribe.go:284]:
```suggestion
func (p *Plugin) getSubscriptionTemplatesByID(instanceID, templateID types.ID) (*SubscriptionTemplate, error) {
```
Shouldn't templateID be also `types.ID` everywhere?

File: server/subscribe.go [hunk 6]
```diff
@@ -1204,6 +1215,12 @@ func (p *Plugin) httpEditSubscriptionTemplates(w http.ResponseWriter, r *http.Re
 		return code, err
 	}
 
+	_ = p.API.SendEphemeralPost(mattermostUserID, &model.Post{
+		UserId:    p.getConfig().botUserID,
+		ChannelId: subscriptionTemplate.ChannelID,
+		Message:   fmt.Sprintf("Jira subscription template, \"%v\", was updated by %v", subscriptionTemplate.Name, connection.DisplayName),
```
Comment (manojmalik20) [inline server/subscribe.go:1221]:
```suggestion
		Message:   fmt.Sprintf("Jira subscription template, %q, was updated by %s", subscriptionTemplate.Name, connection.DisplayName),
```

File: server/subscribe.go [hunk 7]
```diff
@@ -1229,6 +1246,12 @@ func (p *Plugin) httpCreateSubscriptionTemplate(w http.ResponseWriter, r *http.R
 		return code, err
 	}
 
+	_ = p.API.SendEphemeralPost(mattermostUserID, &model.Post{
+		UserId:    p.getConfig().botUserID,
+		ChannelId: subscriptionTemplate.ChannelID,
+		Message:   fmt.Sprintf("Jira subscription template, \"%v\", was added by %v", subscriptionTemplate.Name, connection.DisplayName),
```
Comment (manojmalik20) [inline server/subscribe.go:1252]: Same.

File: server/subscribe.go [hunk 8]
```diff
@@ -1261,6 +1294,12 @@ func (p *Plugin) httpDeleteSubscriptionTemplate(w http.ResponseWriter, r *http.R
 		return code, err
 	}
 
+	_ = p.API.SendEphemeralPost(mattermostUserID, &model.Post{
+		UserId:    p.getConfig().botUserID,
+		ChannelId: subscriptionTemplate.ChannelID,
+		Message:   fmt.Sprintf("Jira subscription template, \"%v\", was removed by %v", subscriptionTemplate.Name, connection.DisplayName),
```
Comment (manojmalik20) [inline server/subscribe.go:1300]: Same.

File: server/subscribe.go (hunk 5 above)
Comment (Kshitij-Katiyar) [reply server/subscribe.go:284]: I need the template Id here as a string to save it in KVstore. We usually have keys for KV store as string that's why i have taken it as a string.

Comment (manojmalik20) [reply server/subscribe.go:284]: But isn't `types.ID` also a string?

**Review by manojmalik20 — APPROVED** [review]

**Pull request closed without merging**